ebusd-2.1.x/de/vaillant/08.v3x00.csv

```diff
@@ -0,0 +1,18 @@
+# type (r[1-9];w;u),circuit,name,[comment],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment
+#,v3x00,VR39,0020139898,,,,,,,,,,
```

Comment:
what kind of circuit is this? is this a mixer or room controller? please note here

Reply:
https://www.wolf-online-shop.de/Vaillant-VR39-Zusatzmodul-0020139898::38909.html

This is an interface ebus <> ProBus to connect old burners to ebus.

Reply:
so it is targeting burners only and is only capable of talking to burners?

Reply:
The VR39 is an adapter for older burners, which do not have ebus but (in case of vaillant devices) an interface labelled "X6", to provide an ebus interface for ebus controllers.
The X6 interface is a proprietary vaillant serial interface, but there is some information public, e.g. https://old.ethersex.de/index.php/Vaillant_X6_Schnittstelle

I have a VR39 as well and would like to assist to get it supported. How can I provide useful data?
I have a Vaillant VC 136-E heater, currently controlled by an VRC410 (via the pin 7-8-9 interface), VR39 installed and an odroid sbc with the ebus adapter 3 installed. A VRC470 is available, but not installed, because I guess it is advantageous to analyse the VR39 alone first?

I attached some logs of the ebusd with and without the provided vr39 definition by rousseldenis. I'm unsure what's next. Grab/dump I guess, but I need some hints. Shall I install and configure the VRC470 and then run the ebusd logging for a while? Which options are necessary to be helpful?

[ebusd_scan_VR39_with_first_definition.log](https://github.com/john30/ebusd-configuration/files/7837797/ebusd_scan_VR39_with_first_definition.log)
[ebusd_scan_VR39_with_first_definition_with_lograw.log](https://github.com/john30/ebusd-configuration/files/7837798/ebusd_scan_VR39_with_first_definition_with_lograw.log)
[ebusd_scan_VR39_without_definition.log](https://github.com/john30/ebusd-configuration/files/7837799/ebusd_scan_VR39_without_definition.log)
[ebusd_scan_VR39_without_definition_with_lograw.log](https://github.com/john30/ebusd-configuration/files/7837801/ebusd_scan_VR39_without_definition_with_lograw.log)

Reply:
Great @Silverstar !

Indeed, I've succeeded recognized VR39 with this first config file.

But, I didn't know how to "discover" features.

As I have currently some failures like 'No Oil', I suppose we can retrieve them through VR39. The link you gave is a first clue.

Reply:
I guess we can only get some values when the corresponding events happen and we are able to correlate the messages.
But hopefully most of the messages are the same as other heaters with ebus, because I don't think the ebus controllers have an extra "VR39 mode" but rather talk to it like to a native ebus heater. So we will "just" have to discover the VR39 specific features, and all the "standard" stuff can come from the other definitions.
But that is something @john30 has to tell us. Thank you in advance!

Reply:
I've been curious and eager to have some progress, so I was fiddling around:
rm 08.v3x00.csv
cp 08.bai.csv 08.v3x.csv
tested, it loads fallback bai.308523.inc and I performed a readout of all values via ip/data/?required
I edited the file so it loads bai.0010015600.inc and did the readout again.
So, many data points work (so, show no error on read) but I have to check if they return valid values.
[ebusd_vr39_test_with_bai.308523.inc.txt](https://github.com/john30/ebusd-configuration/files/7849092/ebusd_vr39_test_with_bai.308523.inc.txt)
[ebusd_vr39_test_with_bai.0010015600.inc.txt](https://github.com/john30/ebusd-configuration/files/7849093/ebusd_vr39_test_with_bai.0010015600.inc.txt)

Reply:
> I've been curious and eager to have some progress, so I was fiddling around)

Hello Silverstar,
Is it working as expected? I have an old Vaillant also without the eBus on the board and therefore also need the VR39.
What is your experience with it? 

Best regards
D

Reply:
@hallengreenn Hey, well I didn't have the time yet to sit in front of the burner and check the display against the numbers I read, but some do already look promising (time, date and outside temperature for example).
So some will be copy & paste. But some will require manual work. Next step I want to do is install the vrc470 and check what I can grab. Maybe that will force some communication which I can "read".

Reply:
> @hallengreenn Hey, well I didn't have the time yet to sit in front of the burner and check the display against the numbers I read, but some do already look promising (time, date and outside temperature for example). So some will be copy & paste. But some will require manual work. Next step I want to do is install the vrc470 and check what I can grab. Maybe that will force some communication which I can "read".

Hey again,

So I have tested the VR39 on my EcoTec Classic VC126/2-C.
I tried copying other bai, with various success.
Many of those datapoints are incorrect. My focus have been around the 'flame' & 'ignition'. And those values haven't been 'on' while my display is showing them as active. 
So my guess is we have to discover these manually.

@john30 how do we start the manual progress of discovering this device? 
From getting the values of the VC39, to translate it into something read-able. 
From there it should be 'easy' enough to add into a csv.

Reply:
Long time, no see, I'm working again on getting the VR39 working with ebusd. I had to set up a new pi zero as the former sbc stopped working. I have now running it together with the VRC470 and I'm testing various configurations, which return useful values and want to strip them from those which return ERR.

@rousseldenis and @hallengreenn can you tell me how your VR39 are identifying themself on the ebus? I guess, that depending on the burner they are attached to, they will have different registers available and so we might need different configs for different burners with the vr39, so like vr39.vc136e, vr39.vc126, ... to address that.

Mine is "MF=Vaillant;ID=V3x00;SW=0118;HW=9902" on the VC 136 E and I'm wondering if there is a difference in SW / HW if on another burner (or is there any other value like a product ID?). If not, we might have to find another way to distinguish them - if necessary.

Reply:
@Silverstar The VR39 appears exactly the same as yours.

In fact, I think that interface should be unique on the bus.

I use only the commands from my VRC regulation and I haven't explored no more VR39 possible features

Reply:
Thanks for your reply. I think that we might get some info about the burner only from the VR39, not from the VRC, so I'm interested to achieve some of the possibilities. I have tested all of the 08.bai... files against the VR39, made a list which items returned an error and which got a value back. Then I will check them against my burner which of those values match, because I already found differences in the different files, returning different values for the same named datapoint.

Which VRC do you have do you use?

Reply:
I have a VRC 450 (I picked the 470 and adapted it, not found the time to do a PR for that)